### ISSUE TITLE
String argument's reset button not shown if re-input same value

### ISSUE THREAD

**Issue by davidlatwe** [mottosso/qargparse.py#7]
### Problem

1. If the default value of the string argument is "A"
2. Type "B" and press Enter, reset button shown
3. Press reset, value fallback to default "A"
4. Type "B" again, reset button not shown

Is this an expected behavior ?

I found that it's because the `self._previous` attribute is holding the previous input from user, but not fallback to default when reset pressed.

https://github.com/mottosso/qargparse.py/blob/2dcdc179d874a36c2f00607ca57695220c2f6e29/qargparse.py#L464-L470

So it's actually :

|  Steps | `_previous` |
| -- | -- |
| Default value of the string argument is "A" | `_previous` == "A" |
| Type "B" and press Enter, reset button shown | `_previous` == "B", thanks to `String.onEditingFinished` |
| Press reset, value fallback to default "A" | `_previous` == "B", because `String.write` triggered by reset did not update it |
| Type "B" again, reset button not shown | `current` == `self._previous`, changed signal not emit |

**Comment by mottosso** [comment]
I'm actually using this right now, and have noticed issues with the reset button. Not just for String, but Enum and Integer too. I think there's an issue in logic somewhere..

With PySide2.

![qargparsebug](https://user-images.githubusercontent.com/2152766/89050978-72305700-d34b-11ea-93b5-f9712dc0d18f.gif)

**Comment by mottosso** [comment]
Come to think of it, the QSpinBox issue is probably something sillier, like the default max value being 99, and it thus never reaching the default value again. :P

**Comment by mottosso** [comment]
Yes, that was it. There's a `max=9999` one could pass to the `add_argument` or `Integer` arguments. In that case, my issues are probably separate from the issue you're seeing.
